Briefcase can't launch if storage folder is moved or deleted

If you launch Briefcase (master) via gradle and you've moved the cache.ser file, it won't run. 

```
./gradlew clean run
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details
Parallel execution is an incubating feature.

> Task :compileJava
Note: /Users/yanokwa/Documents/Work/Nafundi/Code/yanokwa/odk-briefcase/src/org/opendatakit/briefcase/export/ExportEvent.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :run FAILED
Failed to launch Briefcase GUI2018-06-06 10:36:55,248 [main] ERROR o.o.a.p.BaseFormParserForJavaRosa - Failed to launch GUI

java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /Users/yanokwa/Desktop/ODK Briefcase Storage/cache.ser
        at org.opendatakit.briefcase.reused.UncheckedFiles.createFile(UncheckedFiles.java:133)
        at org.opendatakit.briefcase.util.FormCache.from(FormCache.java:52)
        at java.base/java.util.Optional.map(Optional.java:254)
        at org.opendatakit.briefcase.ui.MainBriefcaseWindow.<init>(MainBriefcaseWindow.java:92)
        at org.opendatakit.briefcase.ui.MainBriefcaseWindow.launchGUI(MainBriefcaseWindow.java:79)
        at org.opendatakit.briefcase.Launcher.lambda$main$1(Launcher.java:75)
        at org.opendatakit.common.cli.Cli.lambda$run$5(Cli.java:132)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at org.opendatakit.common.cli.Cli.run(Cli.java:132)
        at org.opendatakit.briefcase.Launcher.main(Launcher.java:79)
Caused by: java.nio.file.NoSuchFileException: /Users/yanokwa/Desktop/ODK Briefcase Storage/cache.ser
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:215)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:369)
        at java.base/java.nio.file.Files.createFile(Files.java:640)
        at org.opendatakit.briefcase.reused.UncheckedFiles.createFile(UncheckedFiles.java:131)
        ... 9 common frames omitted
```

--- a/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
+++ b/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
@@ -27,6 +27,7 @@ import java.awt.Component;
 import java.awt.Toolkit;
 import java.awt.event.WindowAdapter;
 import java.awt.event.WindowEvent;
+import java.nio.file.Files;
 import java.util.HashMap;
 import java.util.Map;
 import javax.swing.ImageIcon;
@@ -88,6 +89,7 @@ public class MainBriefcaseWindow extends WindowAdapter {
 
     BriefcasePreferences appPreferences = BriefcasePreferences.appScoped();
     FormCache formCache = appPreferences.getBriefcaseDir()
+        .filter(Files::exists)
         .map(FormCache::from)
         .orElse(FormCache.empty());
 

--- a/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
+++ b/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
@@ -91,6 +91,8 @@ public class MainBriefcaseWindow extends WindowAdapter {
 
     BriefcasePreferences appPreferences = BriefcasePreferences.appScoped();
     Optional<Path> briefcaseDir = appPreferences.getBriefcaseDir().filter(Files::exists);
+    if (!briefcaseDir.isPresent())
+      appPreferences.unsetStorageDir();
     FormCache formCache = briefcaseDir
         .map(FormCache::from)
         .orElse(FormCache.empty());

--- a/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
+++ b/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java
@@ -28,8 +28,10 @@ import java.awt.Toolkit;
 import java.awt.event.WindowAdapter;
 import java.awt.event.WindowEvent;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 import javax.swing.ImageIcon;
 import javax.swing.JFrame;
 import javax.swing.JTabbedPane;
@@ -88,8 +90,8 @@ public class MainBriefcaseWindow extends WindowAdapter {
     AnnotationProcessor.process(this);
 
     BriefcasePreferences appPreferences = BriefcasePreferences.appScoped();
-    FormCache formCache = appPreferences.getBriefcaseDir()
-        .filter(Files::exists)
+    Optional<Path> briefcaseDir = appPreferences.getBriefcaseDir().filter(Files::exists);
+    FormCache formCache = briefcaseDir
         .map(FormCache::from)
         .orElse(FormCache.empty());
 

